### PR TITLE
fix(#1545): dispatch Event.Resize on all window resizes, requires nvim 0.9+

### LINKED ISSUE #1545: Resize event not dispatched when resizing using mouse
In my init.vim script I subscribe to the resize event, if I use `:NvimTreeResize` the subscribed function gets called, but if I resize with the mouse it doesn't.

https://user-images.githubusercontent.com/35864488/186197406-3a3d54c5-176c-4295-90af-9bd987de25b6.mp4

From the above video you can see that if I use the command, the tabs above follow as well, but if I resize using the mouse they don't.
You can also see how I listen to the event.

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #1545

@wlh320 @Tcholly I'd be most grateful if you tested this fix:

```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 1545-resize-events
```

### CODE REVIEW

File: lua/nvim-tree.lua [hunk 1]
```diff
@@ -337,6 +338,21 @@ local function setup_autocommands(opts)
       end,
     })
   end
+
+  -- TODO #1545 remove similar check from view.resize
+  if vim.fn.has "nvim-0.9" == 1 then
+    create_nvim_tree_autocmd("WinResized", {
+      callback = function()
+        if vim.v.event and vim.v.event.windows then
```
Comment (gegoune) [inline lua/nvim-tree.lua:346]: Not sure if it makes any difference but fallback receives an event afair.

Comment (alex-courtis) [reply lua/nvim-tree.lua:346]: Do you mean the fallback in view.lua?

Sending multiple events would be undesirable...

Comment (gegoune) [reply lua/nvim-tree.lua:346]: > Do you mean the fallback in view.lua?
> 
> 
> 
> Sending multiple events would be undesirable...

No, meant that you should be able to do 

```lua
callback = function(event)
```

Comment (alex-courtis) [reply lua/nvim-tree.lua:346]: Of course, I'll give that a go.

Comment (alex-courtis) [reply lua/nvim-tree.lua:346]: Unfortunately no. The `v:event` data is still not passed to the callback. I hope they add that at some point...

**Comment by alex-courtis** [comment]
nudge @wlh320 @Tcholly

I'd be most grateful if you tested this fix:

```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 1545-resize-events
```

**Pull request merged**